### PR TITLE
Fix for Issue #76

### PR DESCRIPTION
**Pull request by neutronc**
Refactored setPriority in StockRound with an argument indicating a kind of Action. Action currently cant derived directly..
More refactoring necessary perhaps at a later stage. But in the Moment the setPriority Routine gets overwritten in 1835 and 1825 for priority changes only after Buy actions not after sell or other actions.
Check needed for StartCompany...

### CODE REVIEW

File: src/main/java/net/sf/rails/game/specific/_1825/StockRound_1825.java [hunk 1]
```diff
@@ -81,5 +81,14 @@ protected void gameSpecificChecks (PortfolioModel boughtFrom, PublicCompany comp
...
     }
 
+	@Override
+	protected void setPriority(String string) {
+		if (string.equals("BuyCert")) {
+		super.setPriority(string);
+		}
+		else {
+			return;
+		}
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/specific/_1825/StockRound_1825.java]: This part is not required

File: src/main/java/net/sf/rails/game/specific/_1825/StockRound_1825.java [hunk 2]
```diff
@@ -81,5 +81,14 @@ protected void gameSpecificChecks (PortfolioModel boughtFrom, PublicCompany comp
 
     }
 
+	@Override
+	protected void setPriority(String string) {
+		if (string.equals("BuyCert")) {
+		super.setPriority(string);
+		}
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/specific/_1825/StockRound_1825.java]: Formatting

File: src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java [hunk 3]
```diff
@@ -271,5 +271,15 @@ protected boolean checkIfSplitSaleOfPresidentAllowed() {
...
+
+	@Override
+	protected void setPriority(String string) {
+		if (string.equals("BuyCert")) {
+		super.setPriority(string);
+		}
+		else {
+			return;
+		}
+	}
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java]: See above

File: src/main/java/net/sf/rails/game/specific/_1837/StockRound_1837.java [hunk 4]
```diff
@@ -208,7 +208,7 @@ protected boolean mergeCompanies(PublicCompany minor, PublicCompany major) {
                 setSellObligationLifted(major);
             }
 
-            setPriority();
+            setPriority("MergeCompany");
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/specific/_1837/StockRound_1837.java]: Maybe replace with an enum?

File: src/main/java/net/sf/rails/game/specific/_1825/StockRound_1825.java (hunk 1 above)
Comment (neutronc) [reply src/main/java/net/sf/rails/game/specific/_1825/StockRound_1825.java]: technically you are correct...

File: src/main/java/net/sf/rails/game/specific/_1837/StockRound_1837.java (hunk 4 above)
Comment (neutronc) [reply src/main/java/net/sf/rails/game/specific/_1837/StockRound_1837.java]: Quicker to be handled than the string mechanics ?

Comment (madoar) [reply src/main/java/net/sf/rails/game/specific/_1837/StockRound_1837.java]: What do you mean by `quicker`? In my opinion it is more secure. When you use a `String` you have the possible issue of a misspelling, enums don't allow for such errors.

In theory it should also be possible to not do these checks at all by using a class comparison which is really ugly...

**Comment by neutronc** [comment]
I'll have a look in the Stockround SetPriority handling tomorrow...
And probably redo that. If anyone else in the meantime has spare time you are more than welcome to do like madoar suggests..

**Pull request merged**